relay-general/src/store/transactions/processor.rs

```diff
@@ -138,62 +142,67 @@ impl<'r> TransactionsProcessor<'r> {
-                    "end timestamp is smaller than start timestamp",
-                ));
-            }
-            Ok((start, end))
-        }
-        (_, None) => {
-            // This invariant should be already guaranteed for regular error events.
-            Err(ProcessingAction::InvalidTransaction(
-                "timestamp hard-required for transaction events",
-            ))
+    fn validate_transaction(&self, event: &mut Event) -> ProcessingResult {
+        self.validate_timestamps(event)?;
+
+        let Some(trace_context) = event.context_mut::<TraceContext>() else {
+            return Err(ProcessingAction::InvalidTransaction(
+                "missing valid trace context",
+            ));
+        };
+
+        if trace_context.trace_id.value().is_none() {
+            return Err(ProcessingAction::InvalidTransaction(
+                "trace context is missing trace_id",
+            ));
         }
-        (None, _) => {
-            // XXX: Maybe copy timestamp over?
-            Err(ProcessingAction::InvalidTransaction(
-                "start_timestamp hard-required for transaction events",
-            ))
+
+        if trace_context.span_id.value().is_none() {
+            return Err(ProcessingAction::InvalidTransaction(
+                "trace context is missing span_id",
+            ));
         }
-    }
-}
 
-fn validate_transaction(event: &mut Event) -> ProcessingResult {
-    validate_timestamps(event)?;
+        trace_context.op.get_or_insert_with(|| "default".to_owned());
+        Ok(())
+    }
 
-    let Some(trace_context) = event.context_mut::<TraceContext>() else {
-        return Err(ProcessingAction::InvalidTransaction(
-            "missing valid trace context",
-        ));
-    };
+    /// Returns start and end timestamps if they are both set and start <= end.
+    fn validate_timestamps(&self, transaction_event: &Event) -> ProcessingResult {
+        match (
+            transaction_event.start_timestamp.value(),
+            transaction_event.timestamp.value(),
+        ) {
+            (Some(start), Some(end)) => {
+                if end < start {
+                    return Err(ProcessingAction::InvalidTransaction(
+                        "end timestamp is smaller than start timestamp",
+                    ));
+                }
 
-    if trace_context.trace_id.value().is_none() {
-        return Err(ProcessingAction::InvalidTransaction(
-            "trace context is missing trace_id",
-        ));
-    }
+                if let Some(ref range) = self.timestamp_range {
+                    let Some(timestamp) = UnixTimestamp::from_datetime(end.into_inner()) else {
+                        return Err(ProcessingAction::InvalidTransaction("invalid unix timestamp"));
+                    };
+                    if !range.contains(&timestamp) {
+                        return Err(ProcessingAction::InvalidTransaction(
+                            "timestamp is out of the valid range for metrics",
```

Comment:
nit: I'd remove the reason ("...for metrics"), if the metrics requirements change it's easy to forget about this and have an inaccurate error message.

Reply:
This range is taken directly from the metrics aggregator config, that's why this message mentions metrics.